aten/src/ATen/native/TensorProperties.cpp

```diff
@@ -53,10 +53,19 @@ Tensor & detach_(Tensor & self) {
 }
 
 Tensor contiguous(const Tensor & self) {
-  if (self.is_contiguous()) {
+  return contiguous(self, MemoryFormat::Contiguous);
+}
+
+Tensor contiguous(const Tensor & self, MemoryFormat memory_format) {
+  if (self.is_contiguous(memory_format)) {
     return self;
   }
-  return self.clone();
+  auto result = at::empty_like(self);
+  if (memory_format == MemoryFormat::ChannelsLast) {
```

Comment:
It feels safer to me to use a `switch` on the enum here, so that when new memory formats get added, we can have the compiler warn us about non-exhaustive checks.